### ISSUE TITLE
Support for NK NV41 SD card reader in sys-usb and other non-dom0 qubes

### ISSUE THREAD

**Issue by UndeadDevel** [QubesOS/qubes-issues#8484]
### The problem you're addressing (if any)
On the NK (not NovaCustom) NV41 the SD-card reader does seem to work properly, but only if "left alone", i.e. not attached to any qube. This is on R4.1.2.

SD cards can be inserted and will be detected and can then be attached as block devices to other qubes, so the basic functionality is there.

However, attaching the PCI device to, e.g. sys-usb and then inserting the same SD-card has no effect, i.e. there is no popup notification or other sign that there is a new block device attached and I cannot find the device anywhere, so it seems that it doesn't actually get detected. FWIW I'm using debian-12 as the template for sys-usb. I also tried with a custom "sys-sd", but same result. I also tried rebooting, but no difference. I'm not even seeing anything in dmesg or journalctl when inserting the SD-card while the reader is attached to sys-usb.

I realize that compared to generic USB devices, an SD-card is much less problematic security wise, but it would still be nice to have this working properly, with attachment of the PCI-device to sys-usb or other cube possible, especially since this is on a model that lots of Qubes-fans will consider buying.

### The solution you'd like
Attaching the SD-card reader to sys-usb or a custom "sys-sd" allows continued normal usage of the device on NV41.

### The value to a user, and who that user might be
Security benefit to all buying one of the currently most modern Qubes-supporting laptops.

**Comment by UndeadDevel** [comment]
Update on this: I tried with an sd-qube that had Fedora 38 as template, but it also fails...looking into dmesg tells:
`sdhci-pci: probe of 0000:00:06.0 failed with error -22`

FWIW the SD-Card reader is listed fully in lspci, but without a "kernel driver in use" line...I'm also seeing a usb hub with 6 ports (it's a 6-in-1 reader), which is probably also the reader.

I tried some random solutions I found online but none worked...my best bet is that it's a kernel issue.

**Comment by DemiMarie** [comment]
Could also be a Xen limitation.

**Comment by UndeadDevel** [comment]
No improvement unfortunately in Q4.2rc5.

**Comment by DemiMarie** [comment]
I don’t think this is a duplicate.  #2055 is about not assigning card readers to dom0 by default.  Users can override this decision manually, though.  This issue is saying that assigning card readers to another VM does not work.

I could see these issues being related, but they are not identical. The fix might be the same, though.  In particular, this would be the case if the device is initially assigned to dom0 and winds up in an inconsistent state after being deassigned.

@UndeadDevel can you try `rd.qubes.hide_pci=` with the appropriate information for your card reader?  That should prevent the device from being assigned to dom0 in the first place.

**Comment by ben-grande** [comment]
I have an NV56. I assigned the SD/MMC Card Controller to sys-usb. Got the same error on dmesg: `sdhci-pci: probe of 0000:00:06.0 failed with error -22`.


```
# lspci | grep SD
2f:00.0 SD Host controller: O2 Micro, Inc. OZ711 SD/MMC Card Reader Controller (rev 01)

# qvm-pci ls | grep SD
dom0:00_1c.5-00_00.0  SD Host controller: O2 Micro, Inc. OZ711 SD/MMC Card Reader Controller

# qvm-pci at -p sys-usb dom0:00_1c.5-00_00.0
dom0:00_1c.5-00_00.0  SD Host controller: O2 Micro, Inc. OZ711 SD/MMC Card Reader Controller                                   sys-usb (attached)
```
Interesting, the `lspci` and `qvm-pci` gives different addresses for the device, maybe it switched? But only for the SD card, for other devices in `sys-usb`, I can find them with the same identifier on `qvm-pci` and `lspci`.

Added to `/etc/grub/default` and did `grub2-mkconfig -o /boot/grub2`. Verified next boot the cmdline:
```
# cat /proc/cmdline

rd.qubes.hide_pci=2f:00.0
```

Still same error on `sys-usb` dmesg.

**Comment by ben-grande** [comment]
Every reboot, the `lspci` ID stays the same and the `qvm-pci` also stays the same, but different than each other.
